subsys/bluetooth/mesh/crypto.h

```diff
@@ -153,3 +155,12 @@ int bt_mesh_pub_key_gen(void);
 const uint8_t *bt_mesh_pub_key_get(void);
 
 int bt_mesh_dhkey_gen(const uint8_t *pub_key, const uint8_t *priv_key, uint8_t *dhkey);
+
+
+int bt_mesh_encrypt_new(const mesh_key_t key, const uint8_t plaintext[16], uint8_t enc_data[16]);
+int bt_mesh_ccm_encrypt_new(const mesh_key_t key, uint8_t nonce[13],
+			const uint8_t *plaintext, size_t len, const uint8_t *aad,
+			size_t aad_len, uint8_t *enc_data, size_t mic_size);
+int bt_mesh_ccm_decrypt_new(const mesh_key_t key, uint8_t nonce[13],
```

Comment:
This is a temporary solution. I had to add them since `bt_mesh_ccm_decrypt`, `bt_mesh_ccm_encrypt` are used both by network and transport layers. I did changes only for network derivated keys. When all keys will migrate on this approach only one variant of functions will be left. All functions ...._new finally should go away,

Reply:
done